python/src/aiconfig/editor/client/src/components/prompt/PromptMenuButton.tsx

```diff
@@ -10,13 +11,27 @@ const useStyles = createStyles(() => ({
 
 export default memo(function PromptMenuButton({
   promptId,
+  prompt,
+  onDeleteOutput,
   onDeletePrompt,
 }: {
   promptId: string;
+  prompt: ClientPrompt;
   onDeletePrompt: (id: string) => void;
+  onDeleteOutput: (promptId: string) => void;
 }) {
   const { classes } = useStyles();
 
+  const deleteOutput = useCallback(
+    async () => await onDeleteOutput(promptId),
+    [promptId, onDeleteOutput]
+  );
+
+  const deletePrompt = useCallback(
+    async () => await onDeletePrompt(promptId),
+    [promptId, onDeletePrompt]
+  );
```

Comment:
This is a front-end thing that Ryan taught me. Since `PromptMenuButton` is already wrapped with a `memo` component, it's actually more overhead for us to wrap these sub-functions in a `useCallback()` hook. Here `useCallback()` checks for changes to any dependencies, but in our case that's simply changes to the props, which is already memoized in the `PromptMenuButton` component itself, so in this case it adds overhead. Would be better to either:

1. Keep inlined
2. Keep separate but don't wrap with `useCallback()`

Reply:
Oh just noticed as well but we should just construct the `onDeletePrompt` and `onDeleteOutput` callbacks in the parent, since the promptId is known there. Then `promptId` wouldn't even need to be passed, just the callbacks. Similarly that would make PromptMenuButton memoization actually work as intended since the inline function currently prevents memoization from working (new function prop each time). So in `PromptsContainer` we could have
```
const promptId = prompt._ui.id;

const onDeletePrompt = props.onDeletePrompt;
const onDeleteOutput = props.onDeleteOutput;

const deletePrompt = useCallback(() => onDeletePrompt(promptId), [onDeletePrompt, promptId]);
const deleteOutput = useCallback(() => onDeleteOutput(promptId), [onDeleteOutput, promptId]);
```
And they can be passed through the props to PromptMenuButton